src/Authentication/Authentication/TokenCache/WindowsTokenCache.cs

```diff
@@ -4,27 +4,62 @@
 
 namespace Microsoft.Graph.PowerShell.Authentication.TokenCache
 {
+    using System.IO;
     using System.Security.Cryptography;
+    /// <summary>
+    /// Contains methods to store, get and encrypt access tokens using Windows DPAPI in token cache file.
+    /// </summary>
     internal static class WindowsTokenCache
     {
         /// <summary>
-        /// Encrypts the passed buffer using Windows DPAPI.
+        /// Gets a decrypted token from the token cache.
         /// </summary>
-        /// <param name="buffer">A <see cref="byte[]"/> to encrypt.</param>
-        /// <returns>An encrypted <see cref="byte[]"/>.</returns>
-        public static byte[] EncryptToken(byte[] buffer)
+        /// <param name="appId">The app/client id..</param>
+        /// <returns>A decrypted access token.</returns>
+        public static byte[] GetToken(string appId)
         {
-            return ProtectedData.Protect(buffer, null, DataProtectionScope.CurrentUser);
+            if (!Directory.Exists(Constants.TokenCacheDirectory))
```

Comment:
>if (!Directory.Exists(Constants.TokenCacheDirectory)) [](start = 12, length = 53)

doesn't look like you need this check: [https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory?view=netcore-3.1](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory?view=netcore-3.1)